ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,239 @@
+
+	[JsonInclude]
+	[Required]
+	public override Version Version => typeof(MonitoringPlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));
+
+	private bool Enabled => (ASF.GlobalConfig?.IPC ?? false) && (Config?.Enabled ?? MonitoringConfig.DefaultEnabled);
+
+	private const string MeterName = nameof(ArchiSteamFarm);
+
+	private const string MetricNamePrefix = "asf";
+
+	private Meter? Meter;
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			static builder => {
+				builder.AddPrometheusExporter(static config => config.ScrapeEndpointPath = "/Api/metrics");
+				builder.AddMeter("Microsoft.AspNetCore.Hosting");
+				builder.AddMeter("Microsoft.AspNetCore.Server.Kestrel");
+				builder.AddMeter(MeterName);
+
+				builder.AddView(
+					"http.server.request.duration",
+					new ExplicitBucketHistogramConfiguration {
+						Boundaries = [0, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10]
+					}
+				);
+			}
+		);
+	}
+
+	public void OnConfiguringApplication(IApplicationBuilder app) {
+		ArgumentNullException.ThrowIfNull(app);
+
+		if (!Enabled) {
+			return;
+		}
+
+		app.UseEndpoints(static builder => builder.MapPrometheusScrapingEndpoint());
+	}
+
+	[JsonInclude]
+	private MonitoringConfig? Config { get; set; }
+
+	public Task OnASFInit(IReadOnlyDictionary<string, JsonElement>? additionalConfigProperties = null) {
+		if (additionalConfigProperties == null) {
+			return Task.CompletedTask;
+		}
+
+		MonitoringConfig? config = null;
+
+		foreach ((string configProperty, JsonElement configValue) in additionalConfigProperties) {
+			try {
+				config = configProperty switch {
+					nameof(Monitoring) => configValue.Deserialize<MonitoringConfig>(),
+					_ => config
+				};
+			} catch (Exception e) {
+				ASF.ArchiLogger.LogGenericException(e);
+				ASF.ArchiLogger.LogGenericWarning(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+				return Task.CompletedTask;
+			}
+		}
+
+		config ??= new MonitoringConfig();
+
+		if (!Enabled) {
+			ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+			return Task.CompletedTask;
+		}
+
+		Config = config;
+
+		ASF.ArchiLogger.LogGenericInfo(string.Format(CultureInfo.CurrentCulture, Strings.PluginInitializedAndEnabled, nameof(MonitoringPlugin)));
+
+		return Task.CompletedTask;
+	}
+
+	public override Task OnLoaded() {
+		Utilities.WarnAboutIncompleteTranslation(Strings.ResourceManager);
+
+		Meter = new Meter(MeterName, Version.ToString());
+
+		_ = Meter.CreateObservableGauge(
+			$"{MetricNamePrefix}_memory_usage",
+			static () => GC.GetTotalMemory(false) / 1024,
+			description: "Current memory usage of ASF"
+		);
```

Comment:
You can use `OpenTelemetry.Instrumentation.Runtime` package to import most of the .NET runtime metrics in order to not reimplement them yourself.

Reply:
This was done to be in line with what `/Api/ASF` reports. Not sure we want all of .NET runtime.

Reply:
@JustArchi your opinion?

Reply:
I'd rather limit plugin only to what is custom / not-implemented elsewhere. Yes, `/Api/ASF` has some other information ASF-ui is using, but it doesn't have any alternative available from us to show that information, so it's justified - you do have alternatives.

Reply:
Let's do it then.